cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -3657,8 +3657,21 @@ macro(build_grpc)
   # Yuck, see https://stackoverflow.com/a/45433229/776560
   string(REPLACE ";" "|" GRPC_PREFIX_PATH_ALT_SEP "${GRPC_CMAKE_PREFIX}")
 
+  # Negate warnings that gRPC cannot build under
+  # See https://github.com/grpc/grpc/issues/29417
+  set(GRPC_C_FLAGS
+      "${EP_C_FLAGS} -Wno-attributes -Wno-format-security -Wno-unknown-warning-option")
+  set(GRPC_CXX_FLAGS
+      "${EP_CXX_FLAGS} -Wno-attributes -Wno-format-security -Wno-unknown-warning-option")
+  string(REPLACE "-Wformat-security" "" GRPC_C_FLAGS "${GRPC_C_FLAGS}")
+  string(REPLACE "-Wformat-security" "" GRPC_CXX_FLAGS "${GRPC_CXX_FLAGS}")
```

Comment:
Do we need them? I think that `-Wno-format-security` wins with `-Wformat-security -Wno-format-security`. (`-Wformat-security` is ignored.)